### PR TITLE
Pop up login dialog on launch

### PR DESCRIPTION
**Pull request by wayne-chen**
- MS[#18308](https://highfidelity.manuscript.com/f/cases/18308/Add-Pop-up-to-Encourage-Login-if-Not-Already-Logged-In)
### Desktop mode
- Launch Interface
- If not signed in, login dialog should show up after 3 seconds after launch
### Steam (Desktop)
- Launch Interface through Steam
- If not signed in, Steam login dialog should show up after 3 seconds after launch
### HMD mode
- Launch Interface
- If not signed in, login dialog should show up after 3 seconds after launch inside of the tablet
    - tablet should spawn if not spawned already
### Steam (HMD)
- Launch Interface through Steam
- If not signed in, Steam login dialog should show up after 3 seconds after launch

"Keep me signed in" checkbox should allow users to stay signed in upon exit and re-launch

The login dialog should now look exactly like [this](https://user-images.githubusercontent.com/10539521/45238603-3d1e4580-b297-11e8-8452-15219284f0d7.png) in Desktop mode
The login dialog in HMD mode should now look like [this](https://user-images.githubusercontent.com/10539521/45238603-3d1e4580-b297-11e8-8452-15219284f0d7.png) in HMD mode without the close button

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12786/HighFidelity-Beta-PR13961-695f6a8.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12786/HighFidelity-Beta-PR13961-695f6a8.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12786/HighFidelity-Beta-Interface-PR13961-695f6a8.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12786/HighFidelity-Beta-PR13961-695f6a8.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12786/HighFidelity-Beta-Interface-PR13961-695f6a8.exe

**Comment by JeffClinton** [comment]
Transferring my comments from the prior PR:
Initial notes:
-When launching into HMD mode, the tablet appears scaled down and out of reach. This occurs on a fresh install, and after quitting from hmd mode and relaunching (apologies for the sub-standard screenshot)
-Login prompt is skipped if you launch into hmd mode, then immediately change to desktop mode from the desktop menu (display>desktop)
-This build appears to be crashing when quitting from hmd mode. Launch, enter hmd mode (display>oculus rift), quit (file>quit), relaunch (I chose to Visit Sandbox from the system tray).

tablet scale

**Comment by JeffClinton** [comment]
Note: Removed prior comments.
The crashing from HMD mode is present in master.
Unable to repro the tablet scaling issue.
I have no must fix issues at this time.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12850/HighFidelity-Beta-PR13961-5ece8d8.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12850/HighFidelity-Beta-PR13961-5ece8d8.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12850/HighFidelity-Beta-Interface-PR13961-5ece8d8.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12850/HighFidelity-Beta-PR13961-5ece8d8.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12850/HighFidelity-Beta-Interface-PR13961-5ece8d8.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12852/HighFidelity-Beta-PR13961-030ee13.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12852/HighFidelity-Beta-PR13961-030ee13.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12852/HighFidelity-Beta-Interface-PR13961-030ee13.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12852/HighFidelity-Beta-PR13961-030ee13.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12852/HighFidelity-Beta-Interface-PR13961-030ee13.exe

**Comment by JeffClinton** [comment]
Note:  Here are the test cases.

Login Prompt (Desktop)
https://highfidelity.testrail.net/index.php?/cases/view/25283
Login Prompt - Steam (Desktop)
https://highfidelity.testrail.net/index.php?/cases/view/25285
Login Prompt (HMD)
https://highfidelity.testrail.net/index.php?/cases/view/25286
Login Prompt - Steam (HMD)
https://highfidelity.testrail.net/index.php?/cases/view/25287

**Comment by JeffClinton** [comment]
Currently testing

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12858/HighFidelity-Beta-PR13961-6bcf6ea.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12858/HighFidelity-Beta-Interface-PR13961-6bcf6ea.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12858/HighFidelity-Beta-PR13961-6bcf6ea.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12858/HighFidelity-Beta-Interface-PR13961-6bcf6ea.exe

File: libraries/networking/src/UserActivityLogger.cpp [hunk 1]
```diff
@@ -38,10 +38,10 @@ void UserActivityLogger::logAction(QString action, QJsonObject details, JSONCall
     if (_disabled.get()) {
         return;
     }
-    
+
```
Comment (danteruiz) [inline libraries/networking/src/UserActivityLogger.cpp]: I am glad that you cleaned up the whitespace in this file but you didn't change anything this file.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12865/HighFidelity-Beta-PR13961-aa62a89.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12865/HighFidelity-Beta-PR13961-aa62a89.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12865/HighFidelity-Beta-Interface-PR13961-aa62a89.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12865/HighFidelity-Beta-PR13961-aa62a89.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12865/HighFidelity-Beta-Interface-PR13961-aa62a89.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12869/HighFidelity-Beta-PR13961-800df1b.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12869/HighFidelity-Beta-PR13961-800df1b.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12869/HighFidelity-Beta-Interface-PR13961-800df1b.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12869/HighFidelity-Beta-PR13961-800df1b.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12869/HighFidelity-Beta-Interface-PR13961-800df1b.exe

**Comment by JeffClinton** [comment]
FAIL @Conklin94122

Unable to complete log out.  Choosing file>logout will bring up the sign in prompt, but will not log out the user.  You can switch to another account successfully, but you cannot remain in a logged out state.

Steps to repro:
1. Launch interface
2. Sign into account
3. Attempt to sign out of account (file>logout)

[hifi-log.txt](https://github.com/highfidelity/hifi/files/2372336/hifi-log.txt)
[hifi-log_2018-09-11_11.51.13.txt](https://github.com/highfidelity/hifi/files/2372337/hifi-log_2018-09-11_11.51.13.txt)
[hifi-log_2018-09-11_11.51.48.txt](https://github.com/highfidelity/hifi/files/2372338/hifi-log_2018-09-11_11.51.48.txt)



logout

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12876/HighFidelity-Beta-PR13961-e7aef92.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12876/HighFidelity-Beta-PR13961-e7aef92.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12876/HighFidelity-Beta-Interface-PR13961-e7aef92.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12876/HighFidelity-Beta-PR13961-e7aef92.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12876/HighFidelity-Beta-Interface-PR13961-e7aef92.exe

**Comment by JeffClinton** [comment]
Currently testing

**Comment by JeffClinton** [comment]
PASS @Conklin94122

File: libraries/networking/src/UserActivityLogger.cpp (hunk 1 above)
Comment (danteruiz) [reply libraries/networking/src/UserActivityLogger.cpp]: no its fine

File: interface/src/ui/LoginDialog.h [hunk 2]
```diff
@@ -27,7 +27,10 @@ class LoginDialog : public OffscreenQmlDialog {
 
     LoginDialog(QQuickItem* parent = nullptr);
 
+    virtual ~LoginDialog();
```
Comment (danteruiz) [inline interface/src/ui/LoginDialog.h]: Why is this virtual? I don't see any overrides in this PR.

Comment (wayne-chen) [reply interface/src/ui/LoginDialog.h]: You're right. I'll change this.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12889/HighFidelity-Beta-PR13961-9ee21c9.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12889/HighFidelity-Beta-PR13961-9ee21c9.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12889/HighFidelity-Beta-Interface-PR13961-9ee21c9.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12889/HighFidelity-Beta-PR13961-9ee21c9.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12889/HighFidelity-Beta-Interface-PR13961-9ee21c9.exe

**Pull request merged**